tests/languages/pygrep_test.py

```diff
@@ -23,9 +24,21 @@ def some_files(tmpdir):
         ("h'q", 1, "f3:1:with'quotes\n"),
     ),
 )
-def test_main(some_files, cap_out, pattern, expected_retcode, expected_out):
-    ret = pygrep.main((pattern, 'f1', 'f2', 'f3'))
+def test_main(
+    some_files,
+    cap_out,
+    pattern,
+    expected_retcode,
+    expected_out,
+    negate,
+):
+    if not negate:
+        ret = pygrep.main((pattern, 'f1', 'f2', 'f3'))
+    else:
+        ret = pygrep.main((pattern, 'f1', 'f2', 'f3', '--negate'))
```

Comment:
[don't put logic in tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html), write a separate test for this clearly demonstrating the use case